### DISCUSSION TITLE
Zeppelin task plugin fails for Zeppelin which enables authentication

### DISCUSSION THREAD

**Discussion by shangzijun666** [apache/dolphinscheduler#13682] [Q&A]
zeppelin task submit failed with error java.lang.Exception: Please login first @EricGao888

**Comment by EricGao888** [comment]
> zeppelin task submit failed with error java.lang.Exception: Please login first @EricGao888

@shangzijun666 Hello, currently we haven't figured out a suitable solution for integration with Zeppelin authentication. Therefore, you have to disable Zeppelin login if u want to use DS zeppelin task plugin. BTW, if u have any good idea about how to integrate, feel free to leave a message. Thx : )

**Reply by shangzijun666** [reply]
Is it possible to determine whether login authentication is required before submitting the post request, and then add POST http://[ZeppelinServer]:[ZeppelinPort]/api/login

**Reply by shangzijun666** [reply]
curl -i --data 'userName=***&password=***' -X POST http://[ZeppelinServer]:[ZeppelinPort]/api/login

**Reply by EricGao888** [reply]
We could add several params for zeppelin task plugin, such as username, password, etc. But the problem is that this approach is insecure and zeppelin user credentials may get leaked easily.

**Reply by shangzijun666** [reply]
make sense , 
thank you

**Reply by shangzijun666** [reply]
Can you submit a shell task to login to Zeppelin before submitting the Zeppelin task, and set the workgroup of both tasks to be the same

**Comment by davidzollo** [comment]
English is needed

**Reply by shangzijun666** [reply]
ok
